Python/Product/VSCode/AnalysisVsc/Program.cs

```diff
@@ -51,5 +60,44 @@ internal static class Program {
             }
 #endif
         }
+
+        private sealed class SingleThreadSynchronizationContext : SynchronizationContext, IDisposable {
+            private readonly ConcurrentQueue<Tuple<SendOrPostCallback, object>> _queue = new ConcurrentQueue<Tuple<SendOrPostCallback, object>>();
+            private readonly UIService _ui;
+            private readonly ManualResetEventSlim _workAvailable = new ManualResetEventSlim(false);
+            private readonly CancellationTokenSource _cts = new CancellationTokenSource();
+
+            public SingleThreadSynchronizationContext(UIService ui) {
+                _ui = ui;
+                Task.Run(() => QueueWorker());
+            }
+
+            public override void Post(SendOrPostCallback d, object state) {
+                _queue.Enqueue(new Tuple<SendOrPostCallback, object>(d, state));
+                _workAvailable.Set();
+            }
+
+            public void Dispose() {
+                _cts.Cancel();
+            }
+
+            private void QueueWorker() {
+                while(true) {
+                    _workAvailable.Wait(_cts.Token);
+                    if(_cts.IsCancellationRequested) {
+                        break;
+                    }
+                    if(_queue.TryDequeue(out var t)) {
```

Comment:
`while (_queue.TryDequeue(out var t))`

Reply:
No, I don't want to change the outer loop. I meant that we'd like to drain the queue and wait only after it is empty.

Reply:
OK